### ISSUE TITLE
[leo_object_storage] Improve concurrency of read operations

### ISSUE THREAD

**Issue by mocchira** [leo-project/leofs#467]
#450 has fixed part of this problem but there is still some room to improve concurrency more such as 
 adopting readers : writers = N : 1 model.

**Comment by yosukehara** [comment]
@mocchira  I've fixed this issue and done UT and IT. So I'd like to ask you to test the latest LeoFS v1.2.20-dev.

**Comment by mocchira** [comment]
As we discussed while ago,
Hard limits should be implemented for safe.
The lower one must be 1.
Since the appropriate higher limit depends on environments,
We may need some tests on common test cases/hardware to decide that.

**Comment by yosukehara** [comment]
We've checked this feature as below:
* [write-process/obj-container : read-process/obj-container = 1 : 1](https://github.com/leo-project/notes/blob/master/leofs/benchmark/leofs/1.2/1.2.20/front/20160311_image_f2m_r8w2_30min_1/README.md)
* [write-process/obj-container : read-process/obj-container = 1 : 3](https://github.com/leo-project/notes/blob/master/leofs/benchmark/leofs/1.2/1.2.20/front/20160315_image_f2m_r8w2_30min_1/README.md)
* [write-process/obj-container : read-process/obj-container = 1 : 10](https://github.com/leo-project/notes/blob/master/leofs/benchmark/leofs/1.2/1.2.20/front/20160315_image_f2m_r8w2_30min_2/README.md)

In conclusion, LeoFS' read performance is increased by 20%.

**Comment by windkit** [comment]
We got some issues in previous tests (Data Set is not loaded fully beforehand)

Here would be the updated results
https://github.com/leo-project/notes/tree/master/leofs/benchmark/leofs/1.2/1.2.20/front/20160316_image_f4m_r95w5_nocache_30min_1
For comparison, result from 1.2.16 is also uploaded
https://github.com/leo-project/notes/tree/master/leofs/benchmark/leofs/1.2/1.2.16/front/20160316_image_f4m_r95w5_nocache_30min_1
